### PR TITLE
interfaces: make findSnapdPath smarter

### PR DESCRIPTION
**Pull request by mvo5**
The current code that finds the snapd binary to read the build-id
for the system-key generation is not very smart and hardcodes
"core". This PR makes it smarter and it will now also find the
snapd snap which means the build-id is valid.

### CODE REVIEW

File: interfaces/system_key.go [hunk 1]
```diff
@@ -73,19 +73,21 @@ type systemKey struct {
...
 	// find the right snapdPath by looking if we are re-execing or not
-	exe, err := os.Readlink("/proc/self/exe")
+	exe, err := osReadlink("/proc/self/exe")
 	if err != nil {
 		return "", err
 	}
 
 	if strings.HasPrefix(exe, dirs.SnapMountDir) {
-		return filepath.Join(dirs.SnapMountDir, "core/current/usr/lib/snapd/snapd"), nil
+		prefix := strings.Split(exe, "/usr/")[0]
```
Comment (zyga) [inline interfaces/system_key.go]: Hmm, what is this doing?

Comment (mvo5) [reply interfaces/system_key.go]: It just a nice way to get the prefix snap dir of the current running exe, here is an example: https://play.golang.org/p/NwyFxVL6OeL

Comment (bboozzoo) [reply interfaces/system_key.go]: Maybe a comment would be useful here:
```
// reexecd' snapd may be coming from either 'core' or 'snapd' snaps, find 
// the local prefix to the snap
// /snap/snapd/123/usr/bin/snap       -> /snap/snapd/123
// /snap/core/234/usr/lib/snapd/snapd -> /snap/core/234
```

File: tests/core18/basic/task.yaml [hunk 2]
```diff
@@ -15,3 +15,8 @@ execute: |
     echo "Check that snapd tools for core18 works"
     snap install test-snapd-tools-core18
     test-snapd-tools-core18.echo hello | MATCH hello
+
+    if python3 -m json.tool < /var/lib/snapd/system-key | grep '"build-id": ""'; then
+        echo "No the build-id of snapd is should not be empty."
```
Comment (zyga) [inline tests/core18/basic/task.yaml]: `"No...` reads weird IMO. Maybe just `"The build-id...`

Comment (mvo5) [reply tests/core18/basic/task.yaml]: Thank you, the grammar is indeed terrible, fixed.

**Pull request merged**